crates/aptos-substreams/src/transaction_converter.rs

```diff
@@ -0,0 +1,275 @@
+        let mut signatures = vec![];
+        if let Some(signature) = &user_request.signature {
+            signature_type = get_signature_type(signature.r#type());
+            if let Some(signature) = &signature.signature {
+                signatures = get_signature_outputs(signature, user_request, info)?;
+            }
+        }
+        let user_txn_output = UserTransactionOutput {
+            hash: info.hash.clone(),
+            sender: user_request.sender.clone(),
+            sequence_number: user_request.sequence_number,
+            max_gas_amount: user_request.max_gas_amount,
+            expiration_timestamp_secs: user_request.expiration_timestamp_secs.clone(),
+            gas_unit_price: user_request.gas_unit_price,
+            timestamp: info.timestamp.clone(),
+            parent_signature_type: signature_type,
+            signatures,
+        };
+        Ok(user_txn_output)
+    } else {
+        bail!("Transaction info missing from Transaction")
+    }
+}
+
+pub fn get_events_output(
+    events: &[Event],
+    transaction_info: &TransactionInfoOutput,
+) -> Result<Vec<EventOutput>> {
+    Ok(events
+        .iter()
+        .map(|event| {
+            let key = event.key.as_ref().map(|k| EventKeyOutput {
+                creation_number: k.creation_number,
+                account_address: k.account_address.clone(),
+            });
+            EventOutput {
+                transaction_hash: transaction_info.hash.clone(),
+                key,
+                sequence_number: event.sequence_number,
+                move_type: String::from(""),
+                data: event.data.clone(),
+            }
+        })
+        .collect())
+}
+
+pub fn get_write_set_changes_output(_input_txn: &Transaction) -> Result<Vec<WriteSetChangeOutput>> {
+    Ok(vec![])
+}
+
+pub fn get_transaction_type(t: TransactionType) -> String {
+    match t {
+        TransactionType::Genesis => String::from("genesis_transaction"),
+        TransactionType::BlockMetadata => String::from("block_metadata_transaction"),
+        TransactionType::User => String::from("user_transaction"),
+        TransactionType::StateCheckpoint => String::from("state_checkpoint_transaction"),
+    }
+}
+
+pub fn get_signature_type(t: SignatureType) -> String {
+    match t {
+        SignatureType::Ed255198 => String::from("ed25519_signature"),
+        SignatureType::MultiEd255198 => String::from("multi_ed25519_signature"),
+        SignatureType::MultiAgent => String::from("multi_agent_signature"),
+    }
+}
+
+pub fn get_signature_outputs(
+    s: &Signature,
+    request: &UserTransactionRequest,
+    info: &TransactionInfoOutput,
+) -> Result<Vec<SignatureOutput>> {
+    match s {
+        Signature::Ed255198(sig) => Ok(vec![parse_single_signature(
+            sig, request, info, true, 0, None,
+        )?]),
+        Signature::MultiEd255198(sig) => {
+            Ok(parse_multi_signature(sig, request, info, true, 0, None)?)
```

Comment:
no need for `OK(something?)` -> if `something` is a `Result`, can just return `something`